ens-improvement-proposals/ensip-XX-platform-source-parameter.md

```diff
@@ -0,0 +1,57 @@
+
+---
+**Authors**: @slobo.eth and Avsa.eth
+
+---
+**Status**: draft
+
+---
+
+### Abstract
+
+There are many reasons that we might want to be able to measure how and where users are registering names. One of them is simply by knowing our own ecosystem and which apps are more popular, but it’s also very important to be able to identify the creators who are helping the ecosystem. Retroactive funding via grants is often better than proactive funding because you can detect what has already worked, instead of stewards having to make bets on what they think would happen. Often in this community we are surprised by things becoming popular that we would never be able to predict. On the other hand, it’s important to make the process transparent, open source and privacy preserving. Havsource data would also facilitate attempts to create referral programs.
+
+### Specification
+
+The ETH Registrar Controller takes as input: name, owner, duration, secret and resolver. The secret parameter is random data that obscures the name before the final step of the registration process is complete. It is 32 bytes of entropy and the ens manager app on app.ens.domains generates it by calling the following:
```

Comment:
Probably better not to specify all the arguments, as this is changing - just specify that there's a secret parameter.